corehq/form_processor/utils/metadata.py

```diff
@@ -125,7 +123,8 @@ def _get_text_attribute(node):
     else:
         value = node
 
-    if not isinstance(value, six.string_types):
-        value = six.text_type(value)
-    soft_assert_type_text(value)
+    if isinstance(value, bytes):
+        value = value.decode('utf-8')
```

Comment:
This function is used to get a text value for "appVersion" and "location". They can be `None`, and it might (maybe, I don't know) be possible that `appVersion` could be an integer. In either case this could change behaviour that other code, for better or worse, expects. I'd encourage you to change this to
```python
    if not isinstance(value, str):
        value = str(value)
```
instead.

Reply:
Urgh, never mind, I've just seen the `elif` below. (I will try a bit harder to remember always to read all the code first, before commenting.) I agree, I don't think this will hurt.